In [17]:
import pandas as pd

In [18]:
starbucks_reviews = pd.read_csv('src/starbucks_reviews.csv')
display(starbucks_reviews.head())

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review                date  
0  Amber and LaDonna at the Starbucks on Southwes...  September 13, 2023  
1  ** at the Starbucks by the fire station on 436...       July 16, 2023  
2  I just wanted to go out of my way to recognize...        July 5, 2023  
3  Me and my friend were at Starbucks and my card...        May 26, 2023  
4  I’m on this kick of drinking 5 cups of warm wa...    January 22, 2023

In [19]:
print(starbucks_reviews.shape)

(850, 5)


In [20]:
starbucks_reviews.dtypes

name         object
location     object
Rating      float64
Review       object
date         object
dtype: object

In [21]:
starbucks_reviews['date'] = pd.to_datetime(starbucks_reviews['date'])
starbucks_reviews.dtypes

name                object
location            object
Rating             float64
Review              object
date        datetime64[ns]
dtype: object

In [22]:
starbucks_reviews['year'] = starbucks_reviews['date'].dt.year
starbucks_reviews['month'] = starbucks_reviews['date'].dt.month
starbucks_reviews['day'] = starbucks_reviews['date'].dt.day
starbucks_reviews['weekday'] = starbucks_reviews['date'].dt.dayofweek

In [23]:
starbucks_reviews.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  Amber and LaDonna at the Starbucks on Southwes... 2023-09-13  2023      9   
1  ** at the Starbucks by the fire station on 436... 2023-07-16  2023      7   
2  I just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  Me and my friend were at Starbucks and my card... 2023-05-26  2023      5   
4  I’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   day  weekday  
0   13        2  
1   16        6  
2    5        2  
3   26        4  
4   22        6

In [30]:
starbucks_reviews["Review"] = starbucks_reviews["Review"].str.lower()

In [31]:
starbucks_reviews.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  amber and ladonna at the starbucks on southwes... 2023-09-13  2023      9   
1  ** at the starbucks by the fire station on 436... 2023-07-16  2023      7   
2  i just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  me and my friend were at starbucks and my card... 2023-05-26  2023      5   
4  i’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   day  weekday  Review_contains_Starbucks  
0   13        2                      False  
1   16        6                      False  
2    5        2                      False  
3   26        4                      False  
4   22        6                      False

In [35]:
starbucks_reviews["Review_contains_Starbucks"] = starbucks_reviews["Review"].str.count("starbucks")

In [36]:
starbucks_reviews.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  amber and ladonna at the starbucks on southwes... 2023-09-13  2023      9   
1  ** at the starbucks by the fire station on 436... 2023-07-16  2023      7   
2  i just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  me and my friend were at starbucks and my card... 2023-05-26  2023      5   
4  i’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   day  weekday  Review_contains_Starbucks  
0   13        2                          1  
1   16        6                          2  
2    5        2                          1  
3   26        4                          1  
4   22        6                          3

In [37]:
starbucks_reviews["wordcount"] = starbucks_reviews["Review"].str.split(" ").apply(lambda x: len(x))
starbucks_reviews.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  amber and ladonna at the starbucks on southwes... 2023-09-13  2023      9   
1  ** at the starbucks by the fire station on 436... 2023-07-16  2023      7   
2  i just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  me and my friend were at starbucks and my card... 2023-05-26  2023      5   
4  i’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   day  weekday  Review_contains_Starbucks  wordcount  
0   13        2                          1         60  
1   16        6                          2        105  
2    5        2                          1         72  
3   26        4                          1         86  
4   22        6                          3         74

In [39]:
starbucks_reviews.describe(include = 'all')

name      location      Rating          Review  \
count     850           850  705.000000             850   
unique    604           633         NaN             814   
top     Linda  New York, NY         NaN  no review text   
freq       13            14         NaN              37   
mean      NaN           NaN    1.870922             NaN   
min       NaN           NaN    1.000000             NaN   
25%       NaN           NaN    1.000000             NaN   
50%       NaN           NaN    1.000000             NaN   
75%       NaN           NaN    2.000000             NaN   
max       NaN           NaN    5.000000             NaN   
std       NaN           NaN    1.397672             NaN   

                                 date         year       month         day  \
count                             850   850.000000  850.000000  850.000000   
unique                            NaN          NaN         NaN         NaN   
top                               NaN          NaN         NaN         NaN   
freq                              NaN          NaN         NaN         NaN   
mean    2016-02-17 12:38:57.882352896  2015.600000    6.897647   15.442353   
min               2000-07-19 00:00:00  2000.000000    1.000000    1.000000   
25%               2012-09-26 18:00:00  2012.000000    4.000000    7.250000   
50%               2016-05-31 00:00:00  2016.000000    7.000000   15.000000   
75%               2019-01-06 06:00:00  2019.000000   10.000000   23.000000   
max               2023-09-14 00:00:00  2023.000000   12.000000   31.000000   
std                               NaN     4.258436    3.400633    8.906376   

           weekday  Review_contains_Starbucks   wordcount  
count   850.000000                 850.000000  850.000000  
unique         NaN                        NaN         NaN  
top            NaN                        NaN         NaN  
freq           NaN                        NaN         NaN  
mean      2.955294                   1.241176   88.525882  
min       0.000000                   0.000000    3.000000  
25%       1.000000                   0.000000   50.000000  
50%       3.000000                   1.000000   86.000000  
75%       5.000000                   2.000000  123.000000  
max       6.000000                   9.000000  226.000000  
std       1.961437                   1.248584   46.477447

The reviews are quite bad in general, as 75% of the people give a rating below 2 out of 5 !
The reviews were written between 2000 and 2023. Most of the reviews are less than 10 years old, as the median value of year is 2016.
The reviews are quite short in general, as the maximal word count is 226.
The first quartile of "count_starbucks" is 0, which means at least 25% of the people don't use the word "starbucks" at all. Then, people citing "starbucks" use it 1 or 2 times in general.